Library/Formula/fstar.rb

```diff
@@ -0,0 +1,91 @@
+class Fstar < Formula
+  desc "Language with a type system for program verification"
+  homepage "https://www.fstar-lang.org/"
+  url "https://github.com/FStarLang/FStar.git",
+    :tag => "v0.9.1.1",
+    :revision => "c41ea17778ff2fef4e75d56604ce1804e1985f33"
+  head "https://github.com/FStarLang/FStar.git"
+
+  depends_on "mono" => :build
+  depends_on "opam" => :build
+  depends_on "ocaml" => :recommended
+  depends_on "z3" => :recommended
+
+  resource "ocamlfind" do
+    url "http://download.camlcity.org/download/findlib-1.5.5.tar.gz"
+    sha256 "aafaba4f7453c38347ff5269c6fd4f4c243ae2bceeeb5e10b9dab89329905946"
+  end
+
+  resource "batteries" do
+    url "https://github.com/ocaml-batteries-team/batteries-included/archive/v2.3.1.tar.gz"
+    sha256 "df778b90fcdb26288d9d92a86e51dd75d6bb7c6e41888c748c7508e8ea58b1d4"
+  end
+
+  def install
+    ENV.deparallelize # Not related to F* : OCaml parallelization
+
+    opamroot = buildpath/"opamroot"
+    ENV["OPAMROOT"] = opamroot
+    ENV["OPAMYES"] = "1"
+    system "opam", "init", "--no-setup"
+    archives = opamroot/"repo/default/archives"
+    modules = []
+    resources.each do |r|
+      r.verify_download_integrity(r.fetch)
+      original_name = File.basename(r.url)
+      cp r.cached_download, archives/original_name
+      modules << "#{r.name}=#{r.version}"
+    end
+    system "opam", "install", *modules
+
+    system "make", "-C", "src/"
+    system "opam", "config", "exec", "--",
+    "make", "-C", "src/ocaml-output/"
+
+    bin.install "src/ocaml-output/fstar.exe"
+
+    (libexec/"stdlib").install Dir["lib/*"]
+    (libexec/"contrib").install Dir["contrib/*"]
+    (libexec/"examples").install Dir["examples/*"]
+    (libexec/"tutorial").install Dir["doc/tutorial/*"]
+    (libexec/"src").install Dir["src/*"]
+    (libexec/"licenses").install "LICENSE-fsharp.txt", Dir["3rdparty/licenses/*"]
+
+    prefix.install_symlink libexec/"stdlib"
+    prefix.install_symlink libexec/"contrib"
+    prefix.install_symlink libexec/"examples"
+    prefix.install_symlink libexec/"tutorial"
+    prefix.install_symlink libexec/"src"
+    prefix.install_symlink libexec/"licenses"
+  end
+
+  def caveats; <<-EOS.undent
+    F* standard library is available in #{prefix}/stdlib:
+    - alias fstar='fstar.exe --include #{prefix}/stdlib --prims prims.fst'
```

Comment:
> Are there cases where an fstar user wouldn't include the stdlib? If this is needed for basic operation, maybe the formula should include an fstar wrapper script instead of asking the user to set up an alias. The alias as written will break upon version upgrades of fstar, too.
> 
Yes, you are right. I didn’t realize that…
I’ll think about it and try to fix that when updating the formula (soon). Thanks ! B.